FlowCrypt/Functionality/Services/Remote Private Key Services/EmailKeyManagerApi.swift

```diff
@@ -68,22 +68,25 @@ actor EmailKeyManagerApi: EmailKeyManagerApiType {
         let decryptedPrivateKeysResponse = try JSONDecoder().decode(DecryptedPrivateKeysResponse.self, from: response.data)
 
         if decryptedPrivateKeysResponse.privateKeys.isEmpty {
-            return .noKeys
+            throw EmailKeyManagerApiError.noKeys
         }
 
-        let privateKeysArmored = decryptedPrivateKeysResponse.privateKeys
-            .map(\.decryptedPrivateKey)
-            .joined(separator: "\n")
-            .data()
-        let parsedPrivateKeys = try await core.parseKeys(armoredOrBinary: privateKeysArmored)
-        // todo - check that parsedPrivateKeys don't contain public keys
-        let areKeysDecrypted = parsedPrivateKeys.keyDetails
-            .compactMap { $0.isFullyDecrypted }
-        if areKeysDecrypted.contains(false) {
-            return .keysAreNotDecrypted
+        var keys: [KeyDetails] = []
+        for privateKey in decryptedPrivateKeysResponse.privateKeys {
+            let parsedPrivateKey = try await core.parseKeys(armoredOrBinary: privateKey.decryptedPrivateKey.data())
+            // todo - check that parsedPrivateKeys don't contain public keys
+            let isKeyDecrypted = parsedPrivateKey.keyDetails
+                .compactMap { $0.isFullyDecrypted }
+            if isKeyDecrypted.contains(false) {
+                throw EmailKeyManagerApiError.keysAreNotDecrypted
+            }
+            if parsedPrivateKey.keyDetails.isEmpty {
+                throw EmailKeyManagerApiError.keysAreInvalid
+            }
```

Comment:
Here, we always expect exactly one key in each `decryptedPrivateKey`. Therefore `count !== 1` would be more appropriate.